metaflow/plugins/aws/batch/batch_decorator.py

```diff
@@ -109,6 +109,10 @@ def myStep(self):
     run_time_limit = None
 
     def __init__(self, attributes=None, statically_defined=False):
+        username = util.get_username()
```

Comment:
If you move L112:115 to step_init() instead, then a much cleaner (prettified) exception gets thrown.